File: android/app/src/main/java/com/vegettable/app/model/PricePrediction.java
package com.vegettable.app.model;

import com.google.gson.annotations.SerializedName;

public class PricePrediction {
    @SerializedName("cropName")
    private String cropName;

    @SerializedName("currentPrice")
    private double currentPrice;

    @SerializedName("predictedPrice")
    private double predictedPrice;

    @SerializedName("changePercent")
    private double changePercent;

    @SerializedName("direction")
    private String direction; // "up" | "down" | "stable"

    @SerializedName("confidence")
    private double confidence;

    @SerializedName("reasoning")
    private String reasoning;

    public String getCropName() { return cropName; }
    public double getCurrentPrice() { return currentPrice; }
    public double getPredictedPrice() { return predictedPrice; }
    public double getChangePercent() { return changePercent; }
    public String getDirection() { return direction; }
    public double getConfidence() { return confidence; }
    public String getReasoning() { return reasoning; }
}

File: android/app/src/main/java/com/vegettable/app/model/PricePrediction.kt
package com.vegettable.app.model

import com.google.gson.annotations.SerializedName

class PricePrediction {
    @SerializedName("cropName")
    val cropName: String? = null

    @SerializedName("currentPrice")
    val currentPrice: Double = 0.0

    @JvmField
    @SerializedName("predictedPrice")
    val predictedPrice: Double = 0.0

    @JvmField
    @SerializedName("changePercent")
    val changePercent: Double = 0.0

    @JvmField
    @SerializedName("direction")
    val direction: String? = null // "up" | "down" | "stable"

    @JvmField
    @SerializedName("confidence")
    val confidence: Double = 0.0

    @JvmField
    @SerializedName("reasoning")
    val reasoning: String? = null
}
